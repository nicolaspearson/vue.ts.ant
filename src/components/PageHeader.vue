<template>
  <header class="header">
    <section class="logo-alt">
      <LogoAlt />
    </section>
    <nav>
      <ul v-if="navItems">
        <li v-for="navItem in navItems" :key="`${navItem.key}`">
          <NavItem :navItem="navItem" />
        </li>
      </ul>
    </nav>
  </header>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import LogoAlt from '@/assets/logo-alt.svg';
import NavItem from '@/components/NavItem.vue';
import INavItem from '@/interfaces/nav-item';

@Component({ components: { LogoAlt, NavItem } })
export default class PageHeader extends Vue {
  private navItems: INavItem[] = [
    {
      key: 1,
      text: 'Home',
      location: {
        name: 'home',
      },
      color: 'default',
    },
    {
      key: 2,
      text: 'About',
      location: {
        name: 'about',
      },
      color: 'default',
    },
  ];
}
</script>

<style scoped lang="scss">
.header {
  @include gradient();
  @include shadow();
  display: grid;
  grid-template-columns: 200px 1fr 1fr;
  align-items: center;
  padding: 0.5em 2em;

  ul {
    display: flex;
    list-style-type: none;
    padding: 0;
    margin-bottom: 0;
  }

  li {
    display: inline-block;
    margin: 0 4px;
  }

  .logo-alt {
    justify-self: start;
    svg {
      transform: scale(0.75);
    }
  }

  h1 {
    color: white;
  }
}
</style>
