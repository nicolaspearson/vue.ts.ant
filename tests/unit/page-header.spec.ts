import { shallowMount } from '@vue/test-utils';

import PageHeader from '@/components/PageHeader.vue';

describe('PageHeader.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(PageHeader, {
      propsData: {},
      stubs: ['LogoAlt', 'NavItem'],
    });
    expect(wrapper).toMatchSnapshot();
  });
});
