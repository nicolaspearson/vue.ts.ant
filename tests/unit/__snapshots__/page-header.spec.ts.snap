// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader.vue matches snapshot 1`] = `
<header class="header">
  <section class="logo-alt">
    <logoalt-stub></logoalt-stub>
  </section>
  <nav>
    <ul>
      <li>
        <navitem-stub navitem="[object Object]"></navitem-stub>
      </li>
      <li>
        <navitem-stub navitem="[object Object]"></navitem-stub>
      </li>
    </ul>
  </nav>
</header>
`;
